src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/replication/ReplicationFailedHandler.java

```diff
@@ -40,6 +40,7 @@
 @Extension
 public class ReplicationFailedHandler extends RunListener<AbstractBuild<?, ?>> {
 
+    // TODO: scooby doo'd here wrt AbstractBuild. What are the implications?
```

Comment:
There is no equivalent to `RunListener.setUpEnvironment` that would work for Workflow builds, nor probably could there be one; the implicit assumption would not hold.

Anyway the comment below shows that the usage here has nothing to do with environment variables, the plugin just wants to kill builds that have this action present, and this is the only apparent API to do that.

I do not really follow what `ReplicationQueueTaskDispatcher` is doing. Under some conditions it adds this action, which would cause the build to fail if it were scheduled, yet then returns `null` from `canRun`, indicating that the build may be scheduled. Mystery.